### ISSUE TITLE
Raspberry Pi / ARM compatibility

### ISSUE THREAD

**Issue by lucasff** [sgan81/apfs-fuse#25]
Hi! Awesome project :)

I'm trying to build it on Raspbian, but I'm getting a bunch of errors on `lzvn_decode.s`. There's a possibility of ARM compatibility?
Thanks!

**Comment by sgan81** [comment]
Yes, that's on the todo-list ... writing the lzvn_decode in C ...

**Comment by kleuter** [comment]
https://github.com/lzfse/lzfse

**Comment by sgan81** [comment]
Well, lzfse is not lzvn. I'm aware of lzfse though, and will introduce it if needed at some point.

Lzvn is needed to decompress some files. It can be commented out though. In Decmpfs.cpp, in the function DecompressFile, just comment out the if blocks for algos 7 and 8. That's the only place lzvn_decode is used.

In the current version, just put an `#if 0` before line 261 and an `#endif` after line 372 of Decmpfs.cpp. Then, remove lzvn_decode.s from the CMakeLists.txt file in the ApfsLib directory.

**Comment by kleuter** [comment]
this project has lzvn support
src/lzvn_decode_base.h                LZVN decoder
src/lzvn_decode_base.c
src/lzvn_encode_base.h                LZVN encoder
src/lzvn_encode_base.c

**Comment by sgan81** [comment]
Oh I see. That will solve the problem 😃

**Comment by sgan81** [comment]
I have now replaced the assembly code with Apple's lzfse library. And my own unsafe decompression code with zlib. So it should now theoretically compile on the Pi without modification.

**Comment by lucasff** [comment]
Man, you're great! I will buy you some beer, for sure!
I'm having now this:
```
apfs-fuse: malloc.c:2406: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.
Aborted
```

Running: `sudo ./apfs-fuse /dev/sda /mnt`

**Comment by sgan81** [comment]
Now it works.

**Comment by tomkoen** [comment]
Can be closed 😄

**Comment by lucasff** [comment]
It works beautifully. Vielen Dank!
